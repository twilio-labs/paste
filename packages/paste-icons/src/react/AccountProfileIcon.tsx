/**
 * This file was automatically generated with @twilio-labs/svg-to-react
 */
import * as React from 'react';
import {UID} from 'react-uid';
import {IconWrapper, IconWrapperProps} from './helpers/IconWrapper';

export interface AccountProfileIconProps extends IconWrapperProps {
  title?: string;
  decorative?: boolean;
}

const AccountProfileIcon: React.FC<AccountProfileIconProps> = ({title, decorative, ...props}) => (
  <IconWrapper {...props}>
    <UID>
      {uid => (
        <svg
          role="img"
          aria-hidden={decorative}
          aria-labelledby={uid}
          height="100%"
          width="100%"
          viewBox="0 0 1024 1024"
        >
          <title id={uid}>{title}</title>
          <path
            fill="currentColor"
            d="M512.346 472.615c-133.908 0-236.308-102.4-236.308-236.307C276.038 102.4 378.438 0 512.346 0s236.308 102.4 236.308 236.308c0 133.907-102.4 236.307-236.308 236.307m0-393.846c-86.646 0-157.538 70.893-157.538 157.539 0 86.646 70.892 157.538 157.538 157.538 86.646 0 157.539-70.892 157.539-157.538 0-86.646-70.893-157.539-157.539-157.539M906.192 1024h-78.769c0-173.292-141.785-315.077-315.077-315.077S197.269 850.708 197.269 1024H118.5c0-220.554 173.292-393.846 393.846-393.846S906.192 803.446 906.192 1024"
          />
        </svg>
      )}
    </UID>
  </IconWrapper>
);

AccountProfileIcon.defaultProps = {
  title: 'Account Profile Icon',
  decorative: true,
};

AccountProfileIcon.displayName = 'AccountProfileIcon';

export {AccountProfileIcon};
