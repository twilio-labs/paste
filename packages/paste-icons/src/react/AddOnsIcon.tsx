/**
 * This file was automatically generated with @twilio-labs/svg-to-react
 */
import * as React from 'react';
import {UID} from 'react-uid';
import {IconWrapper, IconWrapperProps} from './helpers/IconWrapper';

export interface AddOnsIconProps extends IconWrapperProps {
  title?: string;
  decorative?: boolean;
}

const AddOnsIcon: React.FC<AddOnsIconProps> = ({title, decorative, ...props}) => (
  <IconWrapper {...props}>
    <UID>
      {uid => (
        <svg
          role="img"
          aria-hidden={decorative}
          aria-labelledby={uid}
          height="100%"
          width="100%"
          viewBox="0 0 1024 1024"
        >
          <title id={uid}>{title}</title>
          <path
            fill="currentColor"
            d="M567.282 1024V574.464l385.996-220.109v449.536L567.282 1024m55.142-417.638v322.457l275.712-158.464V447.898L622.424 606.362M456.997 1024L71 803.891V267.418L536.408 0l409.19 234.957-488.601 280.627V1024M126.142 770.355l275.712 158.464V483.686l433.408-248.729L536.408 63.283 126.142 299.315v471.04"
          />
        </svg>
      )}
    </UID>
  </IconWrapper>
);

AddOnsIcon.defaultProps = {
  title: 'Add Ons Icon',
  decorative: true,
};

AddOnsIcon.displayName = 'AddOnsIcon';

export {AddOnsIcon};
