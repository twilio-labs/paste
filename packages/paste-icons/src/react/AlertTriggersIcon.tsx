/**
 * This file was automatically generated with @twilio-labs/svg-to-react
 */
import * as React from 'react';
import {UID} from 'react-uid';
import {IconWrapper, IconWrapperProps} from './helpers/IconWrapper';

export interface AlertTriggersIconProps extends IconWrapperProps {
  title?: string;
  decorative?: boolean;
}

const AlertTriggersIcon: React.FC<AlertTriggersIconProps> = ({title, decorative, ...props}) => (
  <IconWrapper {...props}>
    <UID>
      {uid => (
        <svg
          role="img"
          aria-hidden={decorative}
          aria-labelledby={uid}
          height="100%"
          width="100%"
          viewBox="0 0 1024 1024"
        >
          <title id={uid}>{title}</title>
          <path
            fill="currentColor"
            d="M509.453 989.675c-91.702 0-178.309-25.473-254.727-66.229-5.094-5.094-10.189-10.189-10.189-15.283 0-5.095 0-15.284 5.095-20.379l157.93-275.104c5.095-10.189 20.378-15.284 35.662-10.189 45.851 25.473 96.796 25.473 142.647 0 10.189-5.095 25.472-5.095 35.661 10.19l157.931 275.103c5.094 5.095 5.094 15.284 5.094 20.38 0 5.093-5.094 10.188-10.189 15.282-86.607 40.756-173.214 66.23-264.915 66.23M300.577 887.783c127.363 66.23 285.294 66.23 412.657 0L580.776 658.531c-45.851 20.378-101.89 20.378-147.741 0L300.577 887.784m692.856-356.616H677.572c-15.283 0-25.472-10.19-25.472-25.473 0-61.134-30.568-122.269-71.324-147.74-5.094-5.096-10.189-10.19-10.189-15.285 0-5.095 0-15.284 5.095-20.378l157.93-275.105c5.094-10.189 20.378-15.283 35.662-10.189C922.109 123.605 1024 312.103 1024 500.6c-5.095 20.378-15.284 30.568-30.567 30.568m-290.388-50.946H967.96c-10.189-152.836-86.607-300.577-208.875-382.089L626.627 327.386c40.756 35.662 66.229 91.702 76.418 152.836m-361.712 50.946H25.473C10.189 531.168 0 520.978 0 505.695 0 317.197 101.891 128.7 254.726 42.093c10.189-5.095 25.473-5.095 35.662 10.19l157.93 275.103s5.095 10.19 0 15.284c0 5.095-5.094 10.19-10.189 15.284-40.756 25.472-71.323 86.607-71.323 147.741 0 15.283-10.189 25.473-25.473 25.473M50.945 480.222h264.916c10.189-61.134 35.661-117.174 76.418-152.836L259.821 98.133C137.552 179.645 61.134 327.386 50.945 480.223m412.657 0c0-25.473 20.378-45.85 45.851-45.85 25.472 0 45.85 25.471 45.85 45.85 0 25.472-20.378 45.85-45.85 45.85-25.473 5.095-45.851-20.378-45.851-45.85"
          />
        </svg>
      )}
    </UID>
  </IconWrapper>
);

AlertTriggersIcon.defaultProps = {
  title: 'Alert Triggers Icon',
  decorative: true,
};

AlertTriggersIcon.displayName = 'AlertTriggersIcon';

export {AlertTriggersIcon};
