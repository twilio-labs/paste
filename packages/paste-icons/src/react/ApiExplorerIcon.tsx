/**
 * This file was automatically generated with @twilio-labs/svg-to-react
 */
import * as React from 'react';
import {UID} from 'react-uid';
import {IconWrapper, IconWrapperProps} from './helpers/IconWrapper';

export interface ApiExplorerIconProps extends IconWrapperProps {
  title?: string;
  decorative?: boolean;
}

const ApiExplorerIcon: React.FC<ApiExplorerIconProps> = ({title, decorative, ...props}) => (
  <IconWrapper {...props}>
    <UID>
      {uid => (
        <svg
          role="img"
          aria-hidden={decorative}
          aria-labelledby={uid}
          height="100%"
          width="100%"
          viewBox="0 0 1024 1024"
        >
          <title id={uid}>{title}</title>
          <path
            fill="currentColor"
            d="M354.093 1024c-15.835 0-26.392-10.557-26.392-26.392V242.804c0-15.835 10.557-26.392 26.392-26.392 15.835 0 26.392 10.557 26.392 26.392v760.083c0 10.556-10.557 21.113-26.392 21.113m316.701-216.412c-15.835 0-26.392-10.557-26.392-26.392V26.392C644.402 10.557 654.959 0 670.794 0c15.835 0 26.392 10.557 26.392 26.392v760.082c0 10.557-15.835 21.114-26.392 21.114M971.66 1024L665.515 818.144 364.649 1024h-31.67L16.278 807.588 11 781.196V26.392L53.227 5.278l306.144 205.856L660.237 5.278h31.67l316.701 216.413 5.279 21.113v760.083L971.66 1024M63.784 770.639l290.309 200.577 300.866-205.855h31.67l279.752 190.02v-702.02L676.072 52.784 369.928 263.918h-31.67L63.784 73.897v696.742"
          />
        </svg>
      )}
    </UID>
  </IconWrapper>
);

ApiExplorerIcon.defaultProps = {
  title: 'Api Explorer Icon',
  decorative: true,
};

ApiExplorerIcon.displayName = 'ApiExplorerIcon';

export {ApiExplorerIcon};
