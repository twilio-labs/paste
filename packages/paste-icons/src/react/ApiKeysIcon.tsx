/**
 * This file was automatically generated with @twilio-labs/svg-to-react
 */
import * as React from 'react';
import {UID} from 'react-uid';
import {IconWrapper, IconWrapperProps} from './helpers/IconWrapper';

export interface ApiKeysIconProps extends IconWrapperProps {
  title?: string;
  decorative?: boolean;
}

const ApiKeysIcon: React.FC<ApiKeysIconProps> = ({title, decorative, ...props}) => (
  <IconWrapper {...props}>
    <UID>
      {uid => (
        <svg
          role="img"
          aria-hidden={decorative}
          aria-labelledby={uid}
          height="100%"
          width="100%"
          viewBox="0 0 1024 1024"
        >
          <title id={uid}>{title}</title>
          <path
            fill="currentColor"
            d="M179.2 768c0 59.136 64 96.051 115.2 66.509A76.805 76.805 0 00332.8 768c0-59.136-64-96.051-115.2-66.509A76.805 76.805 0 00179.2 768m153.6 256C148.48 1024 0 875.52 0 691.2c0-184.32 148.48-332.8 332.8-332.8 46.08 0 92.16 10.24 138.24 30.72L855.04 0H1024v256h-87.04l-15.36 15.36v87.04h-87.04l-15.36 15.36v87.04h-87.04l-97.28 97.28c20.48 40.96 25.6 87.04 25.6 133.12C665.6 875.52 517.12 1024 332.8 1024m0-619.52c-153.6 0-281.6 128-281.6 281.6 0 153.6 128 286.72 281.6 286.72 153.6 0 281.6-128 281.6-281.6 0-46.08-10.24-87.04-30.72-128l-10.24-15.36L711.68 409.6H768v-56.32l46.08-46.08h56.32v-56.32l46.08-46.08h56.32V51.2h-97.28L476.16 450.56 460.8 435.2c-40.96-20.48-81.92-30.72-128-30.72"
          />
        </svg>
      )}
    </UID>
  </IconWrapper>
);

ApiKeysIcon.defaultProps = {
  title: 'Api Keys Icon',
  decorative: true,
};

ApiKeysIcon.displayName = 'ApiKeysIcon';

export {ApiKeysIcon};
