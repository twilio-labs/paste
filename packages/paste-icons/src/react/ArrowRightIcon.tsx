/**
 * This file was automatically generated with @twilio-labs/svg-to-react
 */
import * as React from 'react';
import {UID} from 'react-uid';
import {IconWrapper, IconWrapperProps} from './helpers/IconWrapper';

export interface ArrowRightIconProps extends IconWrapperProps {
  title?: string;
  decorative?: boolean;
}

const ArrowRightIcon: React.FC<ArrowRightIconProps> = ({title, decorative, ...props}) => (
  <IconWrapper {...props}>
    <UID>
      {uid => (
        <svg
          role="img"
          aria-hidden={decorative}
          aria-labelledby={uid}
          height="100%"
          width="100%"
          viewBox="0 0 1024 1024"
        >
          <title id={uid}>{title}</title>
          <path
            fill="currentColor"
            d="M781.039 466.961L333.056 18.978C320.359 6.281 305.391 0 288.017 0c-17.374 0-32.343 6.281-45.039 18.978C230.415 31.674 224 46.643 224 64.017v895.966c0 17.241 6.415 32.343 18.978 44.906 12.696 12.696 27.798 19.111 45.039 19.111 17.374 0 32.342-6.415 45.039-19.111l447.983-447.984c12.696-12.563 18.978-27.665 18.978-44.905 0-17.374-6.282-32.342-18.978-45.039"
          />
        </svg>
      )}
    </UID>
  </IconWrapper>
);

ArrowRightIcon.defaultProps = {
  title: 'Arrow Right Icon',
  decorative: true,
};

ArrowRightIcon.displayName = 'ArrowRightIcon';

export {ArrowRightIcon};
