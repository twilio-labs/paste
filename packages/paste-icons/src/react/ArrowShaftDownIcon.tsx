/**
 * This file was automatically generated with @twilio-labs/svg-to-react
 */
import * as React from 'react';
import {UID} from 'react-uid';
import {IconWrapper, IconWrapperProps} from './helpers/IconWrapper';

export interface ArrowShaftDownIconProps extends IconWrapperProps {
  title?: string;
  decorative?: boolean;
}

const ArrowShaftDownIcon: React.FC<ArrowShaftDownIconProps> = ({title, decorative, ...props}) => (
  <IconWrapper {...props}>
    <UID>
      {uid => (
        <svg role="img" aria-hidden={decorative} aria-labelledby={uid} width="100%" height="100%" viewBox="0 0 20 20">
          <title id={uid}>{title}</title>
          <path
            fill="currentColor"
            d="M10 13.449V5.5a.5.5 0 111 0v7.949l3.273-3.322a.421.421 0 01.602 0 .437.437 0 010 .611l-4.074 4.135a.421.421 0 01-.602 0l-4.074-4.135a.437.437 0 010-.611.421.421 0 01.602 0L10 13.449z"
            fillRule="evenodd"
          />
        </svg>
      )}
    </UID>
  </IconWrapper>
);

ArrowShaftDownIcon.defaultProps = {
  title: 'Arrow Shaft Down Icon',
  decorative: true,
};

ArrowShaftDownIcon.displayName = 'ArrowShaftDownIcon';

export {ArrowShaftDownIcon};
