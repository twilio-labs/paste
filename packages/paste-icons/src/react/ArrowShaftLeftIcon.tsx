/**
 * This file was automatically generated with @twilio-labs/svg-to-react
 */
import * as React from 'react';
import {UID} from 'react-uid';
import {IconWrapper, IconWrapperProps} from './helpers/IconWrapper';

export interface ArrowShaftLeftIconProps extends IconWrapperProps {
  title?: string;
  decorative?: boolean;
}

const ArrowShaftLeftIcon: React.FC<ArrowShaftLeftIconProps> = ({title, decorative, ...props}) => (
  <IconWrapper {...props}>
    <UID>
      {uid => (
        <svg role="img" aria-hidden={decorative} aria-labelledby={uid} width="100%" height="100%" viewBox="0 0 20 20">
          <title id={uid}>{title}</title>
          <path
            fill="currentColor"
            d="M6.551 9H14.5a.5.5 0 110 1H6.551l3.322 3.273a.421.421 0 010 .602.437.437 0 01-.611 0L5.127 9.801a.421.421 0 010-.602l4.135-4.074a.437.437 0 01.611 0 .421.421 0 010 .602L6.551 9z"
            fillRule="evenodd"
          />
        </svg>
      )}
    </UID>
  </IconWrapper>
);

ArrowShaftLeftIcon.defaultProps = {
  title: 'Arrow Shaft Left Icon',
  decorative: true,
};

ArrowShaftLeftIcon.displayName = 'ArrowShaftLeftIcon';

export {ArrowShaftLeftIcon};
