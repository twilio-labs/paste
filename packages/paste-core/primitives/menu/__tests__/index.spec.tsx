import * as React from 'react';
import {render, fireEvent, screen} from '@testing-library/react';
import axe from '../../../../../.jest/axe-helper';
import {
  useMenuPrimitiveState,
  MenuPrimitive,
  MenuPrimitiveItem,
  MenuPrimitiveButton,
  MenuPrimitiveButtonProps,
  MenuPrimitiveSeparator,
} from '../src';

const PreferencesMenu = React.forwardRef<HTMLButtonElement, MenuPrimitiveButtonProps>((props, ref) => {
  const menu = useMenuPrimitiveState({baseId: 'sub-menu'});
  return (
    <>
      <MenuPrimitiveButton ref={ref} {...menu} {...props} data-testid="example-submenu-trigger">
        Preferences
      </MenuPrimitiveButton>
      <MenuPrimitive {...menu} aria-label="Preferences" data-testid="example-submenu">
        <MenuPrimitiveItem {...menu} data-testid="example-submenu-item">
          Settings
        </MenuPrimitiveItem>
        <MenuPrimitiveItem {...menu} disabled data-testid="example-disabled-submenu-item">
          Extensions
        </MenuPrimitiveItem>
        <MenuPrimitiveSeparator {...menu} />
        <MenuPrimitiveItem {...menu}>Keyboard shortcuts</MenuPrimitiveItem>
      </MenuPrimitive>
    </>
  );
});

const MenuMock: React.FC = () => {
  const menu = useMenuPrimitiveState({baseId: 'menu-example'});
  return (
    <>
      <MenuPrimitiveButton {...menu}>Code</MenuPrimitiveButton>
      <MenuPrimitive {...menu} aria-label="Code">
        <MenuPrimitiveItem {...menu} data-testid="example-menu-item">
          About Visual Studio Code
        </MenuPrimitiveItem>
        <MenuPrimitiveItem {...menu}>Check for Updates...</MenuPrimitiveItem>
        <MenuPrimitiveSeparator {...menu} data-testid="example-menu-separator" />
        <MenuPrimitiveItem {...menu} as={PreferencesMenu} />
      </MenuPrimitive>
    </>
  );
};

describe('Menu Primitive', () => {
  describe('Render', () => {
    it('should render', () => {
      const {asFragment} = render(<MenuMock />);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should render a menu button with aria attributes', () => {
      render(<MenuMock />);
      const renderedMenuButton = screen.getByRole('button');
      expect(renderedMenuButton.getAttribute('aria-haspopup')).toEqual('menu');
      expect(renderedMenuButton.getAttribute('aria-controls')).toEqual('menu-example');
    });

    it('should render a menu', () => {
      render(<MenuMock />);
      const renderedMenu = screen.getByLabelText('Code');
      expect(renderedMenu.getAttribute('role')).toEqual('menu');
      expect(renderedMenu.getAttribute('aria-orientation')).toEqual('vertical');
    });

    it('should render a menu item', () => {
      render(<MenuMock />);
      const renderedMenuItem = screen.getByTestId('example-menu-item');
      expect(renderedMenuItem.getAttribute('role')).toEqual('menuitem');
    });

    it('should render a menu separator', () => {
      render(<MenuMock />);
      const renderedMenuItem = screen.getByTestId('example-menu-separator');
      expect(renderedMenuItem.getAttribute('aria-orientation')).toEqual('horizontal');
      expect(renderedMenuItem.tagName).toEqual('HR');
    });

    it('should render a sub menu trigger', () => {
      render(<MenuMock />);
      const renderedSubMenuTrigger = screen.getByTestId('example-submenu-trigger');
      expect(renderedSubMenuTrigger.getAttribute('role')).toEqual('menuitem');
      expect(renderedSubMenuTrigger.getAttribute('aria-haspopup')).toEqual('menu');
      expect(renderedSubMenuTrigger.getAttribute('aria-expanded')).toEqual('false');
    });

    it('should render a submenu', () => {
      render(<MenuMock />);
      const renderedMenu = screen.getByTestId('example-submenu');
      expect(renderedMenu.getAttribute('role')).toEqual('menu');
      expect(renderedMenu.getAttribute('aria-orientation')).toEqual('vertical');
    });

    it('should render a sub menu item', () => {
      render(<MenuMock />);
      const renderedMenuItem = screen.getByTestId('example-submenu-item');
      expect(renderedMenuItem.getAttribute('role')).toEqual('menuitem');
    });

    it('should render a disabled sub menu item', () => {
      render(<MenuMock />);
      const renderedMenuItem = screen.getByTestId('example-disabled-submenu-item');
      expect(renderedMenuItem.getAttribute('role')).toEqual('menuitem');
      expect(renderedMenuItem.getAttribute('aria-disabled')).toEqual('true');
    });
  });

  // an update to reakit broke these document.activeElements
  // annoying, commenting out for review but will investigate
  // eslint-disable-next-line jest/no-disabled-tests
  describe.skip('interaction', () => {
    it('should control expanded attribute on the button', () => {
      render(<MenuMock />);
      const renderedMenuButton = screen.getByRole('button');
      expect(renderedMenuButton.getAttribute('aria-expanded')).toEqual('false');
      fireEvent.click(renderedMenuButton);
      expect(renderedMenuButton.getAttribute('aria-expanded')).toEqual('true');
      if (document.activeElement != null) {
        fireEvent.keyDown(document.activeElement, {key: 'Escape', code: 'Escape'});
        // eslint-disable-next-line jest/no-conditional-expect
        expect(renderedMenuButton.getAttribute('aria-expanded')).toEqual('false');
      }
    });

    it('should focus first menu item in the menu when open', () => {
      render(<MenuMock />);
      fireEvent.click(screen.getByRole('button'));
      if (document.activeElement != null) {
        // eslint-disable-next-line jest/no-conditional-expect
        expect(screen.getAllByRole('menuitem')[0]).toEqual(document.activeElement);
      }
    });

    it('should move focus to the second menu item in the menu when down arrow pressed', () => {
      render(<MenuMock />);
      fireEvent.click(screen.getByRole('button'));
      if (document.activeElement != null) {
        fireEvent.keyDown(document.activeElement, {key: 'ArrowDown', code: 'ArrowDown'});
        // eslint-disable-next-line jest/no-conditional-expect
        expect(screen.getByText('Check for Updates...')).toEqual(document.activeElement);
      }
    });

    it('should move focus to the first menu item in the menu when down up pressed', () => {
      render(<MenuMock />);
      fireEvent.click(screen.getByRole('button'));
      if (document.activeElement != null) {
        fireEvent.keyDown(document.activeElement, {key: 'ArrowDown', code: 'ArrowDown'});
        fireEvent.keyDown(document.activeElement, {key: 'ArrowUp', code: 'ArrowUp'});
        // eslint-disable-next-line jest/no-conditional-expect
        expect(screen.getByText('About Visual Studio Code')).toEqual(document.activeElement);
      }
    });
  });

  describe('Accessibility', () => {
    it('Should have no accessibility violations', async () => {
      const {container} = render(<MenuMock />);
      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });
  });
});
