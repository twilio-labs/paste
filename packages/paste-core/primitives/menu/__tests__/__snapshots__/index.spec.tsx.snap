// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Primitive Render should render 1`] = `
<DocumentFragment>
  <button
    aria-controls="menu-example"
    aria-expanded="false"
    aria-haspopup="menu"
    type="button"
  >
    Code
  </button>
  <div
    aria-label="Code"
    aria-orientation="vertical"
    data-dialog="true"
    hidden=""
    id="menu-example"
    role="menu"
    style="display: none;"
    tabindex="-1"
  >
    <button
      data-testid="example-menu-item"
      id="menu-example-1"
      role="menuitem"
      tabindex="0"
    >
      About Visual Studio Code
    </button>
    <button
      id="menu-example-2"
      role="menuitem"
      tabindex="-1"
    >
      Check for Updates...
    </button>
    <hr
      aria-orientation="horizontal"
      data-testid="example-menu-separator"
      role="separator"
    />
    <button
      aria-controls="sub-menu"
      aria-expanded="false"
      aria-haspopup="menu"
      data-testid="example-submenu-trigger"
      id="menu-example-3"
      role="menuitem"
      tabindex="-1"
      type="button"
    >
      Preferences
    </button>
    <div
      aria-label="Preferences"
      aria-orientation="vertical"
      data-dialog="true"
      data-testid="example-submenu"
      hidden=""
      id="sub-menu"
      role="menu"
      style="display: none;"
      tabindex="-1"
    >
      <button
        data-testid="example-submenu-item"
        id="sub-menu-1"
        role="menuitem"
        tabindex="0"
      >
        Settings
      </button>
      <button
        aria-disabled="true"
        data-testid="example-disabled-submenu-item"
        disabled=""
        id="sub-menu-2"
        role="menuitem"
        style="pointer-events: none;"
      >
        Extensions
      </button>
      <hr
        aria-orientation="horizontal"
        role="separator"
      />
      <button
        id="sub-menu-3"
        role="menuitem"
        tabindex="-1"
      >
        Keyboard shortcuts
      </button>
    </div>
  </div>
</DocumentFragment>
`;
